test/core/test/mocked.test.ts

```diff
@@ -169,3 +169,29 @@ describe('mocked function which fails on toReturnWith', () => {
 test('streams', () => {
   expect(exportedStream).toBeDefined()
 })
+
+describe('temporary mock implementation', () => {
```

Comment:
There are not enough tests. What if I used `mockImplementationOnce` before this function, for example? Or any other method for that matter